src/main/java/nextstep/optional/ComputerStore.java

```diff
@@ -21,6 +23,14 @@ public static String getVersion(Computer computer) {
     }
 
     public static String getVersionOptional(Computer computer) {
-        return null;
+        return Optional.ofNullable(computer)
+                .filter(c -> Optional.ofNullable(c.getSoundcard())
+                        .filter(soundCard -> Optional.ofNullable(soundCard.getUsb())
+                                .filter(usb -> Optional.ofNullable(usb.getVersion())
+                                        .isPresent())
+                                .isPresent())
+                        .isPresent())
+                .map(c -> c.getSoundcard().getUsb().getVersion())
+                .orElse(UNKNOWN_VERSION);
```

Comment:
```suggestion
        if(computer != null) {
            return computer.getSoundcardUsbVersion();
        }

        return UNKNOWN_VERSION;
```

이 부분을 요구사항에서 제외한 이유는, 단순히 null 체크를 위해 Optional로 감싸는 것이 비효율적이기 때문인 것으로 알고 있어요.

Optional best practice는 아래 링크를 참고해보세요
> http://homoefficio.github.io/2019/10/03/Java-Optional-바르게-쓰기/

---

```c.getSoundcard().getUsb().getVersion()``` 

디미터 원칙 위반입니다. 

> https://github.com/brainbackdoor/refactoring/blob/master/docs/hapter_07_객체간의_기능_이동.md#대리-객체-은폐-hide-delegate

Reply:
유의하겠습니다. 감사합니다 👍